broke-broker/src/main/java/org/dsngroup/broke/broker/channel/handler/ProtocolProcessor.java

```diff
@@ -184,13 +192,42 @@ private MqttSubAckMessage subAck(MqttQoS qos, int packetId, List<MqttQoS> grante
         return new MqttSubAckMessage(mqttFixedHeader, mqttMessageIdVariableHeader, mqttSubAckPayload);
     }
 
+    private void schedulePingReq(Channel channel) {
+        pingReqScheduleFuture = channel.eventLoop().scheduleAtFixedRate(
+                new Runnable(){
+                    @Override
+                    public void run(){
+
+                        int packetId = pingReqPacketIdGenerator.getAndIncrement();
+                        MqttFixedHeader mqttFixedHeader =
+                                new MqttFixedHeader(MqttMessageType.PINGREQ, false, MqttQoS.AT_MOST_ONCE, false, 0);
+                        MqttMessageIdVariableHeader mqttMessageIdVariableHeader =
+                                MqttMessageIdVariableHeader.from(packetId);
+                        MqttPingReqMessage mqttPingReqMessage =
+                                new MqttPingReqMessage(mqttFixedHeader, mqttMessageIdVariableHeader);
+                        // Set the PINGREQ's sendTime
+                        serverSession.setPingReq(packetId);
+                        channel.writeAndFlush(mqttPingReqMessage);
+                    }
+                }, 0, 3, TimeUnit.SECONDS);
+    }
+
```

Comment:
Bad place to put the ping requests here, it's better split out for the unified processing functions in this class.

Reply:
Do you mean to place ```schedulePingReq()``` into a separated class?

The ```schedulePingReq()``` is only called when the connection succeeded (In ```processConnect()``` function). That's why I placed this in the ProtocolProcessor class.

Reply:
In the **semantic** of `schedulePingReq` is related to the `RTT` module, instead of processor. Additionally, the name of `schedulePingReq` is bad, may consider change to `calculateRtt` or anything else, a more clear name.

Reply:
In additional, I think the rtt is not related to `connect`, instead, the `subscribe` is correct.
Anyway, you should figure out the **unified access** in the processing class.

Reply:
Sorry I forgot to write comment for this function. 
This function is used to schedule a periodic PINGREQ message sent to the client. (i.e. send PINGREQ every 3 seconds in the above code)
The calculation of rtt is performed when the corresponding PINGRESP is received.

Reply:
Ok, no need to write comment every where is fined. What I'm actually mean is that, the processor functionalities I think is better unified, i.e. `processConnect`, `processSubscribe`, etc. It is a first class of handling different message types, but the internal complex functionalities should be split out.

Reply:
i.e. you can have a split class like `ProbeScheduler` for integrating rtt and watermark registration.

Reply:
OK, I see. Maybe I'll consider to create an additional ```PingReqScheduler``` class for the ```ProtocolProcessor``` class.

Reply:
@allenlee820202 , I'll merge this first, you can implement it in the next iteration. Also, it's better to split prs and branches for remain non strictly related functionalities to make the progress close to non-blocking.